app/src/main/java/com/example/atm/pinchange/PinChangeActivity.kt

```diff
@@ -0,0 +1,151 @@
+                        if (pinChangeNumber != oldPassword!!) {
+                            if (pinChangeNumber == pinConfirmationNumber) {
+                                db?.details()?.changePassword(pinChangeNumber, mAccountNumber)
+                                this@PinChangeActivity.runOnUiThread { showDialog() }
+
+                            } else {
+                                this@PinChangeActivity.runOnUiThread {
+                                    ConfigUtil().toast(
+                                        this@PinChangeActivity,
+                                        resources.getString(R.string.error_pin_mismatch)
+                                    )
+                                    enterPinToChange.requestFocus()
+                                    enterPinToChange.text?.clear()
+                                    enterPinConfirmation.text?.clear()
+                                }
+                            }
+                        } else {
+                            this@PinChangeActivity.runOnUiThread {
+                                ConfigUtil().toast(
+                                    this@PinChangeActivity,
+                                    resources.getString(R.string.error_already_exist)
+                                )
+                                enterPinToChange.requestFocus()
+                                enterPinToChange.text?.clear()
+                                enterPinConfirmation.text?.clear()
+                            }
+                        }
+                    }
+                } else if (stringPinChange.trim().isEmpty() && stringPinConfirmation.trim()
+                        .isEmpty()
+                ) {
+                    enterPinToChange.requestFocus()
+                    enterPinToChange.error = resources.getString(R.string.error_empty_pin_number)
+                    enterPinConfirmation.error =
+                        resources.getString(R.string.error_empty_pin_number)
+                } else if (stringPinChange.length in 3 downTo 1) {
+                    enterPinToChange.text?.clear()
+                    enterPinConfirmation.text?.clear()
+                    enterPinConfirmation.requestFocus()
+                    enterPinConfirmation.error = resources.getString(R.string.error_invalid_length)
+                } else if (stringPinConfirmation.trim().isEmpty()) {
+                    enterPinConfirmation.requestFocus()
+                    enterPinConfirmation.error =
+                        resources.getString(R.string.error_empty_pin_number)
+                } else if (stringPinChange.trim().isEmpty()) {
+                    enterPinToChange.requestFocus()
+                    enterPinToChange.error = resources.getString(R.string.error_empty_pin_number)
+                }
+            } catch (ex: NumberFormatException) {
+                Log.i(ConfigUtil().logCatTag, ConfigUtil().logCatMessage)
```

Comment:
The log message in the catch block should show the message from the exception, try to avoid the hardcoded messages. The purpose of the log is to get the error message from the exception